src/main/java/org/fundacionjala/coding/nestor/DnaStrand.java

```diff
@@ -0,0 +1,40 @@
+package org.fundacionjala.coding.nestor;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class to resolve complementary DNA.
+ */
+public class DnaStrand {
+  /**
+   * Method make complement.
+   *
+   * @param dna type String.
+   * @return return a string with dna complementary.
+   */
+  public String makeComplement(final String dna) {
+    char[] chars = dna.toCharArray();
+
+    for (int i = 0; i < chars.length; i++) {
+      chars[i] = complement(chars[i]);
+    }
+
+    return new String(chars);
+  }
+
+  /**
+   * Method helper who catch the dna complementary.
+   *
+   * @param c type String.
+   * @return return string which will be a complement.
+   */
+  protected static char complement(final char c) {
+    final Map<Character, Character> dnaComplement = new HashMap<>();
+    dnaComplement.put('A', 'T');
+    dnaComplement.put('C', 'G');
+    dnaComplement.put('G', 'C');
+    dnaComplement.put('T', 'A');
+    return dnaComplement.get(c);
+  }
+}
```

Comment:
This may be refactored like this
```java
class DnaStrand {

  static final Map<Character, Character> DNA_COMPLEMENT = new HashMap<>();
  static {
    DNA_COMPLEMENT.put('A', 'T');
    DNA_COMPLEMENT.put('C', 'G');
    DNA_COMPLEMENT.put('G', 'C');
    DNA_COMPLEMENT.put('T', 'A');
  }
  /**
   * Method make complement.
   *
   * @param dna type String.
   * @return return a string with dna complementary.
   */
  String makeComplement(final String dna) {
    char[] chars = dna.toCharArray();
    for (int i = 0; i < chars.length; i++) {
      chars[i] = DNA_COMPLEMENT.get(chars[i]);
    }
    return new String(chars);
  }
}
```